astropy/io/ascii/cparser.pyx

```diff
@@ -738,6 +741,9 @@ cdef class CParser:
     def set_names(self, names):
         self.names = names
 
+    def get_header_names(self):
+        return self.header_names
+
```

Comment:
I've run into that before too with cdef classes in Cython.  I've never been sure if it's really a "feature" or a bug.  It smells like a bug to me, or at least last time I tried it the resulting behavior was pretty buggy. Never got around to investigating that though.